tcf_website/static/reviews/new_review.js

```diff
@@ -124,8 +124,48 @@ jQuery(function($) {
                 $("#semester").prop("disabled", false);
             });
     });
+
+    // Code for the progress bar
+    $("#reviewtext").keydown(function() {
+        // Used .trim() to remove leading and trailing spaces
+        var review = $("#reviewtext").val().trim();
+        var numberOfWords = countNumberOfWords(review);
+
+        // Set the width of the bar to the what percent of 300 words the current review is
+        $("#reviewprogressbar").width(1000 * (numberOfWords / 300));
+        if (numberOfWords < 100) {
+            $("#progressbarmessage").html("Your review is under 100 words. Aim for 300 or more!");
+        } else if (numberOfWords >= 100 && numberOfWords < 200) {
+            $("#reviewprogressbar").removeClass("progress-bar bg-danger");
+            $("#reviewprogressbar").css("background-color", "#FFDAC1");
+            $("#progressbarmessage").html("Good job getting to 100 words, keep going!");
+        } else if (numberOfWords >= 200 && numberOfWords < 300) {
+            $("#reviewprogressbar").css("background-color", "#fff5ba");
+            $("#progressbarmessage").html("200 words! You're so close to the 300 mark!");
+        } else if (numberOfWords >= 300) {
+            $("#reviewprogressbar").css("background-color", "#B5EAD7");
+            $("#progressbarmessage").html("Thank you for your in depth review. The tCF team and other users appreciate your effort!");
+        }
+    });
 });
 
+// Counts the number of words in a review
+function countNumberOfWords(review) {
+    if (review.length === 0) {
+        return 0;
+    }
+    // If first condition is not met, we have at least one word
+    var count = 1;
+    for (const s of review) {
+        // Chose to count words based on how many spaces there are
+        if (s === " ") {
+            count++;
+        }
```

Comment:
From what I can tell by trying it and the code, this method seems to include punctuation and really any text that might not be a word. This quite clearly wouldn't reflect the actual word count, but the hassle of trying to count words **exclusively** may not be worth the hassle of a rather trivial thing since most of what a user is writing is indeed words.

Reply:
hmmmm good catch. Maybe it's worth having a list of all the characters that would count as contributing to words and only count those?

Reply:
From what I can tell, you seemed to make it so that only actual words are counted as opposed to other special characters, so awesome job again! I'm not sure if there's any more efficient method out there to do it but the speed/performance doesn't seem hindered.